src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_ssl.c

```diff
@@ -433,6 +433,22 @@ int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringR
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
         osStatus = SecTrustEvaluate(trust, &trustResult);
+
+        if (trustResult == kSecTrustResultRecoverableTrustFailure && osStatus == noErr && certificateCount > 1)
+        {
+            // If we get recoverable failure, let's try it again with full anchor list.
```

Comment:
It took me a moment to realize why we were starting from 1, that we already populated anchors with the first item and now we're populating the rest.
```suggestion
            // If we get recoverable failure, let's try it again with full anchor list.
            // We already stored just the first certificate into anchors; now we store the rest.
```

Reply:
updated.